src/Native/System.Security.Cryptography.Native/pal_ecc_import_export.cpp

```diff
@@ -0,0 +1,513 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <openssl/ec.h>
+#include <openssl/objects.h>
+#include "pal_ecc_import_export.h"
+#include "pal_utilities.h"
+
+ECCurveType MethodToCurveType(EC_METHOD* method)
+{
+    if (method == EC_GFp_mont_method())
+        return ECCurveType::PrimeMontgomery;
+
+    if (method == EC_GF2m_simple_method())
+        return ECCurveType::Characteristic2;
+
+    // Fallback including nist curves (optimized EC_GFp_nist_method and others)
+    return ECCurveType::PrimeShortWeierstrass;
```

Comment:
Yeah, I know.  Import/Export may be a bit fragile here; but I'd rather that we fail for new optimizations than have an accelerator for char2-klobitz be added and we call it PrimeShortWeierstrass; or a new kind of thing all together appears and we make the same mistake.